test-suite/output/NumeralNotations.out

```diff
@@ -184,3 +184,41 @@ let v := 4%Zlike in v : Zlike
      : Zlike
 0%Zlike
      : Zlike
+let v := 0%kt in v : ty
+     : ty
+let v := 1%kt in v : ty
+     : ty
+let v := 2%kt in v : ty
+     : ty
+let v := 3%kt in v : ty
+     : ty
+let v := 4%kt in v : ty
+     : ty
+let v := 5%kt in v : ty
+     : ty
+The command has indeed failed with message:
+Cannot interpret this number as a value of type ty
+     = 0%kt
+     : ty
+     = 1%kt
+     : ty
+     = 2%kt
+     : ty
+     = 3%kt
+     : ty
+     = 5%kt
```

Comment:
I would have expected this to be 4%kt (it's should be the result of `Eval cbv in (_ : known_type Set)` if I'm not mistaken)

Reply:
Hm, good point.  I bet there's an issue with the order of typeclass resolution; I'll fix it.

Reply:
Fixed